src/openrct2/interface/console.c

```diff
@@ -359,41 +371,32 @@ double console_parse_double(const utf8 *src, bool *valid) {
     return value;
 }
 
-static void console_update_scroll()
-{
-    /*sint32 lines = 0;
-    sint32 maxLines = ((_consoleBottom - 22 - _consoleTop) / 10) - 1;
-    char *ch = strchr(_consoleBuffer, 0);
-    while (ch > _consoleBuffer && lines < maxLines) {
-        ch--;
-        if (*ch == '\n')
-            lines++;
-    }
-
-    if (*ch == '\n')
-        ch++;
-    _consoleViewBufferStart = ch;*/
-}
-
-void console_scroll(sint32 delta)
+void console_scroll(sint32 linesToScroll)
 {
-    sint32 speed = 3;
+    sint32 speed = abs(linesToScroll);
     sint32 lines = 0;
-    sint32 maxLines = ((_consoleBottom - 22 - _consoleTop) / 10) - 1;
+    sint32 maxLines = console_get_num_visible_lines();
     utf8 *ch = strchr(_consoleBuffer, 0);
     while (ch > _consoleBuffer) {
         ch--;
         if (*ch == '\n')
             lines++;
     }
-    if (delta > 0 && _consoleScrollPos + 1 < (lines - maxLines + 4)) {
+    if (linesToScroll > 0 && _consoleScrollPos + 1 < (lines - maxLines + 4)) {
         _consoleScrollPos = min(_consoleScrollPos + speed, (lines - maxLines + 4));
     }
-    else if (delta < 0 && _consoleScrollPos > 0) {
+    else if (linesToScroll < 0 && _consoleScrollPos > 0) {
         _consoleScrollPos = max(_consoleScrollPos - speed, 0);
     }
 }
 
+// Calculates the amount of visible lines, based on the console size, excluding the input line.
+static sint32 console_get_num_visible_lines()
+{
+    const sint32 lineHeight = 10;
```

Comment:
On second thought, please make this a constant (LINE_HEIGHT) and put it at the top of the file.

Reply:
Well the problem with that is, I believe this is actually the only time "10" is explicitly used for lineheights, as the rest of the draw functionality uses `font_get_line_height` (but 10 was likely used to prevent the need to set the font inside of an input update, as that would likely cause weird effects) . 

Do you still want me to do that? I only added the lineHeight variable as a form of explaining what the 10 was for.

Reply:
Hm, not really sure what is the best course. I merged it anyway, but this may be revisited at some point.